external/java/src/main/java/ch/epfl/dedis/lib/SkipBlock.java

```diff
@@ -94,6 +103,42 @@ public int getIndex(){
         return ret;
     }
 
+    /**
+     * This function checks whether all signatures in the forward-links
+     * are correctly signed by the aggregate public key of the roster
+     *
+     * @return true if the signature is ok.
+     */
+    public boolean verifyForwardSignatures() {
+        List<Point> publics;
+        try {
+            publics = new Roster(this.skipBlock.getRoster()).getServicePublics("Skipchain");
+        } catch (URISyntaxException e) {
+            return false;
+        }
+
+        for (ForwardLink fl : this.getForwardLinks()) {
```

Comment:
Should we have a check that at list one is not empty if the length is above 0 ?

Reply:
I think it's ok if there are no forward links. It might happens when the block is the latest.